src/components/ActiveCard.jsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { activateCard } from '../ducks/crib-sheet';
+import { togglePopup } from '../ducks/dialog';
+import SnippetDeletePrompt from './SnippetDeletePrompt';
+
+const ENABLE_DRAG = 'handle active-crib-card__content';
+const DISABLE_DRAG = 'active-crib-card__content';
+
+class ActiveCard extends React.Component {
+  constructor() {
+    super();
+
+    this.deactivateCard = this.deactivateCard.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.deleteCardPrompt = this.deleteCardPrompt.bind(this);
+    this.editCardText = this.editCardText.bind(this);
+    this.toggleEditBox = this.toggleEditBox.bind(this);
+
+    this.state = {
+      disableSave: true,
+      editBox: false
+    };
+  }
+
+  deactivateCard() {
+    this.props.dispatch(activateCard(null));
+  }
+
+  handleInputChange() {
+    const emptyField = this.inputText.value.length === 0;
+
+    if (!emptyField && this.state.disableSave === true) {
+      this.setState({ disableSave: false });
+    } else if (emptyField && this.state.disableSave === false) {
+      this.setState({ disableSave: true });
+    }
+  }
+
+  editCardText() {
+    if (this.props.preferences.preferences && this.props.preferences.preferences.cribsheet) {
+      const cribCopy = this.props.preferences.preferences.cribsheet.slice();
+      cribCopy[this.props.cardIndex].name = this.inputText.value;
+      this.props.preferences.update({ 'preferences.cribsheet': cribCopy }).save();
```

Comment:
I'm OK with `.save()` not providing success/failure feedback (since this is a non-core feature) but we need to be aware that any potential errors in this bit doesn't cascade out into the main app functinality.

Reply:
To be fair, I am just paranoid after ASM.